ec/src/models/bls12/mod.rs

```diff
@@ -145,35 +143,46 @@ impl<P: Bls12Parameters> PairingEngine for Bls12<P> {
             // r = f^((p^6 - 1)(p^2 + 1))
             r *= &f2;
 
-            // Hard part of the final exponentation is below:
-            // From https://eprint.iacr.org/2016/130.pdf, Table 1
+            // Hard part of the final exponentation:
+            // t[0].CyclotomicSquare(&result)
             let mut y0 = r.cyclotomic_square();
-            y0.conjugate();
-
-            let mut y5 = Self::exp_by_x(r);
-
-            let mut y1 = y5.cyclotomic_square();
-            let mut y3 = y0 * &y5;
-            y0 = Self::exp_by_x(y3);
-            let y2 = Self::exp_by_x(y0);
-            let mut y4 = Self::exp_by_x(y2);
-            y4 *= &y1;
-            y1 = Self::exp_by_x(y4);
-            y3.conjugate();
-            y1 *= &y3;
-            y1 *= &r;
-            y3 = r;
-            y3.conjugate();
-            y0 *= &r;
-            y0.frobenius_map(3);
-            y4 *= &y3;
-            y4.frobenius_map(1);
-            y5 *= &y2;
-            y5.frobenius_map(2);
-            y5 *= &y0;
-            y5 *= &y4;
-            y5 *= &y1;
-            y5
+            // t[1].Expt(&result)
+            let mut y1 = Self::exp_by_x(r);
```

Comment:
Lets label what is `x` and `x/2` in both this function, and as a comment on the function within the file

Reply:
Thanks for adding it! Didn't realize it was literally the bls12 param